crates/starknet/src/starknet/add_deploy_account_transaction.rs

```diff
@@ -0,0 +1,161 @@
+    use crate::starknet::{predeployed, Starknet};
+    use crate::traits::{Accounted, HashIdentifiedMut, StateChanger, StateExtractor};
+    use crate::transactions::deploy_account_transaction::DeployAccountTransaction;
+    use crate::utils::{get_storage_var_address, load_cairo_0_contract_class};
+
+    #[test]
+    fn deploy_account_transaction_should_fail_due_to_low_balance() {
+        let (mut starknet, account_class_hash, _) = setup();
+
+        let transaction = DeployAccountTransaction::new(
+            vec![],
+            2000,
+            vec![],
+            Felt::from(0),
+            account_class_hash,
+            Felt::from(13),
+            DEVNET_DEFAULT_CHAIN_ID.to_felt().into(),
+            Felt::from(0),
+        )
+        .unwrap();
+
+        let (txn_hash, _) = starknet.add_deploy_account_transaction(transaction).unwrap();
+        let txn = starknet.transactions.get_by_hash_mut(&txn_hash).unwrap();
+
+        assert_eq!(txn.status, TransactionStatus::Rejected);
+    }
+
+    #[test]
+    fn deploy_account_transaction_successful_execution() {
+        let (mut starknet, account_class_hash, fee_token_address) = setup();
+
+        let transaction = DeployAccountTransaction::new(
+            vec![],
+            2000,
+            vec![],
+            Felt::from(0),
+            account_class_hash,
+            Felt::from(13),
+            DEVNET_DEFAULT_CHAIN_ID.to_felt().into(),
+            Felt::from(0),
+        )
+        .unwrap();
+
+        // change balance at address
+        let account_address =
+            ContractAddress::try_from(transaction.0.contract_address().clone()).unwrap();
+        let balance_storage_var_address =
+            get_storage_var_address("ERC20_balances", &[account_address.into()]).unwrap();
+        let balance_storage_key =
+            ContractStorageKey::new(fee_token_address, balance_storage_var_address);
+
+        let account_balance_before_deployment = Felt::from(1000000);
+        starknet
+            .state
+            .change_storage(balance_storage_key, account_balance_before_deployment)
+            .unwrap();
+        starknet.state.synchronize_states();
+
+        // get accounts count before deployment
+        let accounts_before_deployment = starknet.state.state.address_to_nonce.len();
+
+        let (txn_hash, _) = starknet.add_deploy_account_transaction(transaction).unwrap();
+        let txn = starknet.transactions.get_by_hash_mut(&txn_hash).unwrap();
+
+        assert_eq!(txn.status, TransactionStatus::AcceptedOnL2);
+        assert_eq!(starknet.state.state.address_to_nonce.len(), accounts_before_deployment + 1);
+        let account_balance_after_deployment =
+            starknet.state.get_storage(balance_storage_key).unwrap();
+
+        assert!(account_balance_before_deployment > account_balance_after_deployment);
+    }
+
+    /// Initializes starknet with 1 account - account without validations
```

Comment:
Also not in the comment that the erc20 is deployed and the gas price is set to 1

Reply:
[2428c54](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/73/commits/2428c54771a90955d5119efbb4cf58e06449a7ae)